src/kbmod/reprojection.py

```diff
@@ -114,10 +127,10 @@ def _reproject_work_unit(work_unit, common_wcs, frame="original"):
         indices = list(np.where(obstimes == time)[0])
         per_image_indices.append(indices)
 
-        science_add = np.zeros(common_wcs.array_shape)
-        variance_add = np.zeros(common_wcs.array_shape)
-        mask_add = np.zeros(common_wcs.array_shape)
-        footprint_add = np.zeros(common_wcs.array_shape)
+        science_add = np.zeros(common_wcs.array_shape, dtype=np.float32)
+        variance_add = np.zeros(common_wcs.array_shape, dtype=np.float32)
+        mask_add = np.zeros(common_wcs.array_shape, dtype=np.float32)
```

Comment:
These three were using the default float64 dtype, and then being cast to float32, which would require another copy in memory. Defining them as float32 to start with saves space and the memory copy.